infra/spawn_pipeline.py

```diff
@@ -85,13 +85,19 @@ def main():
         # mapping between an internal ID and the generate ID
 
         task_id = taskcluster.utils.slugId()
-        task_internal_id = task.pop("ID")
-
+        task_internal_id = task.get("ID")
         if task_internal_id in id_mapping:
             raise ValueError(f"Conflicting IDs {task_internal_id}")
 
+        # add each task id to id_mapping dictionary
+        # before process the task dependencies
         id_mapping[task_internal_id] = task_id
 
+    for task in rendered["tasks"]:
+
```

Comment:
Nit: no newline between `for` and the first line of the block (`task_internal_id = ...`).